File: front-skiller/src/app/data/sonar/FilesStats.ts
/**
 * This class represents the number of files detected on the repository per language.
 */
export class FilesStats {

	/**
	 * @param langage the language counted (Java, TS...)
	 * @param numberOfFiles Number of files detected by Sonar.
	 */
	constructor (
		public language: string,
		public numberOfFiles: number) {}
}

File: back-wibkac/src/main/java/com/tixhi/data/internal/FilesStats.java
package com.tixhi.data.internal;

import java.io.Serializable;

import lombok.Data;

/**
 * <p>
 * Source files statistics per language. 
 * </p>
 *
 * @author Fr&eacute;d&eacute;ric VIDAL
 *
 */
public @Data class FilesStats implements Serializable {

	/**
	 * serialVersionUID for serialization purpose.
	 */
	private static final long serialVersionUID = 6937126988922326182L;

	private String language;
	
	private int numberOfFiles;

	/**
	 * Empty constructor for serialization purpose.
	 */
	public FilesStats() {
	}
	
	/**
	 * Constructor.
	 * @param language the language
	 * @param numberOfFiles the number of files for this language
	 */
	public FilesStats(String language, int numberOfFiles) {
		super();
		this.language = language;
		this.numberOfFiles = numberOfFiles;
	}
	
}
